Docs/change request beta docs (#2608)
Unleash/unleash: website/docs/reference/change-requests.md

---
title: Change requests
---

:::info

The change requests feature was introduced in **Unleash 4.18.0**, and is currently an enterprise only feature in beta

> with a select few customers. If you're an unleash enterprise customer, please contact us if you'd like to try out this 

Reviewer: selected few customers?

feature. 

:::

Feature flagging is a powerful tool, and because it's powerful you sometimes need to practice caution. The ability to have complete control over your production environment comes at the cost of being able to make mistakes in production. Change requests are here to alleviate that fear. Introduced in 4.18.0, you can now group changes in a change request instead of applying them directly to production. This allows you to make changes to multiple feature toggles configuration and status (on/off) grouped together, and applied together at the same time. 

Our goal is developer effeciency, but we also recognize that we have users and customers in highly regulated industries, governed by law and strict requirements. Therefore, we have added a capability to change requests that will allow you to enforce the _4 eyes principle_.

## Change request configuration

> The change request configuration can be set up per project, per environment. This means that you can have different change request configurations for different environments. This is useful if you have different requirements for different environments, for example production and development. The implication of this is that you can't change toggles across projects in the same change request.

Reviewer 2: The change request configuration can be set up per project, per environment. This means that you can have different change request configurations for different environments, such as production and development. This is useful because different environments may have different requirements, so you can customize the change request configuration to fit those requirements. However, this also means that you cannot change toggles across projects in the same change request.

---
title: Change requests
---

:::info

The change requests feature was introduced in **Unleash 4.18.0**, and is currently an enterprise only feature in beta
with a select few customers. If you're an unleash enterprise customer, please contact us if you'd like to try out this 
feature. 

:::

> Feature flagging is a powerful tool, and because it's powerful you sometimes need to practice caution. The ability to have complete control over your production environment comes at the cost of being able to make mistakes in production. Change requests are here to alleviate that fear. Introduced in 4.18.0, you can now group changes in a change request instead of applying them directly to production. This allows you to make changes to multiple feature toggles configuration and status (on/off) grouped together, and applied together at the same time. 

Reviewer 2: ```suggestion
Feature flagging is a powerful tool, and because it's so powerful, you sometimes need to practice caution. The ability to have complete control over your production environment comes at the cost of the potential to make mistakes in production. Change requests were introduced in version 4.18.0 to alleviate this fear. Change requests allow you to group changes together and apply them to production at the same time, instead of applying changes directly to production. This allows you to make multiple changes to feature toggles and their configuration and status (on/off) all at once, reducing the risk of errors in production.
```

Our goal is developer effeciency, but we also recognize that we have users and customers in highly regulated industries, governed by law and strict requirements. Therefore, we have added a capability to change requests that will allow you to enforce the _4 eyes principle_.

## Change request configuration

> The change request configuration can be set up per project, per environment. This means that you can have different change request configurations for different environments. This is useful if you have different requirements for different environments, for example production and development. The implication of this is that you can't change toggles across projects in the same change request.

Reviewer 2: ```suggestion
The change request configuration can be set up per project, per environment. This means that you can have different change request configurations for different environments, such as production and development. This is useful because different environments may have different requirements, so you can customize the change request configuration to fit those requirements. However, this also means that you cannot change toggles across projects in the same change request.
```

![Change request configuration](/img/change-request-configuration.png)

## Change request flow

Once a change request flow is configured for a project and environment, you can no longer directly change the status of a toggle. Instead, you will be asked to put your changes into a draft. The change request flow handles the following scenarios: 

* Updating the status of a toggle in the environment
* Adding a strategy to the feature toggle in the environment
* Updating a strategy of a feature toggle in the environment
* Deleting a strategy from a feature toggle in the environment

The flow can be summarized as follows:

![Change request flow](/img/change-request-flow.png)

Once a change is added to a draft, the draft needs to be completed before another change request can be opened. The draft is personal to the user that created the change request draft, until it is sent for review. Once changes are added to draft, the user will have a banner in the top of the screen indicating that a draft exists. The state of a change request can be one of the following: 

* **Draft** - The change request is in draft mode, and can be edited by the user that created the draft.

> * **In review** - The change request is in review mode, and can be edited by the user that created the draft. If editing ocurrs, all current approvals are revoked

Reviewer: The comment about revoking should be in the Approved item, not In review
